### ISSUE TITLE
Unable to run parallel backtest with neural network models

### ISSUE THREAD

**Issue by Mr-Geekman** [tinkoff-ai/etna-ts#122]
### 🐛 Bug Report

There is an error trying to use `TimeSeriesCrossValidation.backtest` with `n_jobs` > 1 when we use CUDA. 

### Expected behavior

Running parallel backtest.

### Code

Example dataset was taken from datasets for example notebooks.

```python
import pandas as pd
from etna.models.nn import DeepARModel
from etna.datasets.tsdataset import TSDataset
from etna.model_selection import TimeSeriesCrossValidation
from etna.metrics import SMAPE
from etna.transforms import PytorchForecastingTransform


classic_df = pd.read_csv("data/example_dataset.csv")
df = TSDataset.to_dataset(classic_df)
ts = TSDataset(df, freq="D")

HORIZON = 7

transform_deepar = PytorchForecastingTransform(
    max_encoder_length=HORIZON,
    max_prediction_length=HORIZON,
    time_varying_known_reals=["time_idx"],
    time_varying_unknown_reals=["target"],
    target_normalizer=None,
)

model = DeepARModel(max_epochs=3, learning_rate=[0.1], gpus=[0], batch_size=64)
tscv_deepar = TimeSeriesCrossValidation(
    model=model, horizon=HORIZON, metrics=[SMAPE()], n_folds=3, n_jobs=2
)

tscv_deepar.backtest(ts, transforms=[transform_deepar])
```

### Environment

Standard colab environment with installed etna-ts using pip.

### Checklist

- [X] Bug appears at the latest library version
- [X] Bug description added
- [X] Steps to reproduce added
- [X] Expected behavior added

**Comment by julia-shenshina** [comment]
how can we fix it?

**Comment by martins0n** [comment]
We add joblib parameters customization via additional parameter `joblib_params` in #253  for `pipeline.backtest`
```
tscv_deepar.backtest(ts, n_jobs=2, metrics=[SMAPE()], n_folds=2, joblib_params=dict(verbose=11, backend="loky", mmap_mode="c"))
```
Now by default we use `multiprocessing` backend for joblib because of `global` nature of our tslogger.
You can use `loky` or `multiprocessing` with `multiprocessing.get_start_method() == 'spawn'` but results of folds runs are not be logged -- only total results of backtest will be logged (It may be important for experiments tracked by wandb) 


### To sum up:

So in case of working with cuda, you can use parallelization of backtest with  `loky` specification or `multiprocessing` with `multiprocessing.set_start_method('spawn')` specification.

For now we can close the issue. 

You can open it if additional recommendation are needed or you have new proposal.
